File: atcoder/trainings/abc363_c.py
# これで通るんだ..
from more_itertools import distinct_permutations

N, K = map(int, input().split())
S = input().strip()
ans = 0

for arr in distinct_permutations(S):
    ok = True
    for i in range(N - K + 1):
        f = True
        for j in range(K):
            if arr[i + j] != arr[i + K - 1 - j]:
                f = False
                break
        if f:
            ok = False
            break
    if ok:
        ans += 1

print(ans)

File: atcoder/trainings/abc363_c.rb
# frozen_string_literal: true

# https://atcoder.jp/contests/abc363/tasks/abc363_c
#
# ■考えたこと
#
# ■解説
#

N, K = gets.split.map(&:to_i)
S = gets.chomp
slist = Set.new(S.chars.permutation).to_a
ans = 0
slist.each do |arr|
  ok = true
  0.upto(N - K) do |i|
    f = true
    K.times do |j|
      f = false if arr[i + j] != arr[i + K - 1 - j]
    end
    ok = false if f
  end
  ans += 1 if ok
end
puts ans
